plugins/BEdita/Core/src/Command/SearchCommand.php

```diff
@@ -0,0 +1,268 @@
+ * Operations available are:
+ *
+ * - `--reindex`: reindex all objects in the system
+ * - `--index`: index a single object
+ * - `--delete`: delete an object from index
+ * - `--clear`: clear index
+ *
+ * Options available are:
+ *
+ * - `--dry-run`: dry run, do not perform any operation
+ *
+ * Usage examples:
+ *
+ * ```bash
+ * bin/cake search --reindex
+ * bin/cake search --reindex documents,events
+ * bin/cake search --reindex --dry-run
+ * bin/cake search --reindex documents,events --dry-run
+ * bin/cake search --index 25
+ * bin/cake search --index 25 --dry-run
+ * bin/cake search --clear
+ * bin/cake search --clear documents,events
+ * bin/cake search --clear --dry-run
+ * bin/cake search --clear documents,events --dry-run
+ * bin/cake search --delete 25
+ * bin/cake search --delete 25 --dry-run
+ * ```
+ *
+ * @since 5.14.0
+ * @property \BEdita\Core\Model\Table\ObjectsTable $Objects
+ */
+class SearchCommand extends Command
+{
+    /**
+     * Operations available.
+     *
+     * @var string[]
+     */
+    protected $operations = [
+        'reindex',
+        'index',
+        'delete',
+        'clear',
+    ];
+
+    /**
+     * Dry run flag.
+     *
+     * @var bool
+     */
+    protected $dryrun = false;
+
+    /**
+     * @inheritDoc
+     */
+    public function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        $parser = parent::buildOptionParser($parser);
+        $parser->setDescription('Interface to handle search indexes and data.');
+        $parser->addOption('reindex', [
+            'help' => 'Reindex all or multiple objects in the system.',
+            'required' => false,
+        ]);
+        $parser->addOption('index', [
+            'help' => 'Index a single object.',
+            'required' => false,
+        ]);
+        $parser->addOption('delete', [
+            'help' => 'Delete an object from index.',
+            'required' => false,
+        ]);
+        $parser->addOption('clear', [
+            'help' => 'Remove index for all or multiple objects.',
+            'required' => false,
+        ]);
+        $parser->addOption('dry-run', [
+            'help' => 'Dry run, do not perform any operation.',
+            'required' => false,
+        ]);
+
+        return $parser;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(Arguments $args, ConsoleIo $io): ?int
+    {
+        $tmp = array_intersect_key($args->getOptions(), array_flip($this->operations));
+        $operation = empty($tmp) ? '' : (string)array_key_first($tmp);
+        $message = empty($operation) ? $this->getOptionParser()->help() : 'Perform ' . $operation . ' operation';
+        $io->out($message);
+        $this->Objects = $this->fetchTable('Objects');
+        $this->dryrun = $args->getOption('dry-run') !== null;
+
+        return empty($operation) ? Command::CODE_ERROR : $this->{$operation}($args, $io);
```

Comment:
If you like my suggestions per my previous comment, this method might become something like this:

```php
public function execute(Arguments $args, ConsoleIo $io): int
{
    foreach ($this->objectsIterator($args) as $object) {
        $this->doIndexResource($object, $args, $io); // Signature changed to filter adapters, if desired.
    }

    // ...
}

protected function objectsIterator(Arguments $args): iterable
{
    $table = $this->fetchTable('Objects');
    $query = $table->find()->orderAsc($table->aliasField('id'))->limit(200);
    if (!empty($args->getOption('type')) {
        $query = $query->find('type', (array)$args->getOption('type'));
    }
    // More filters...

    $lastId = 0;
    while (true) {
        $q = clone $query;
        $q = $q->where(fn (QueryExpression $exp): QueryExpression => $exp->gt($table->aliasField('id'), $lastId));
        $results = $query->all();
        if ($results->isEmpty()) {
            break;
        }

        foreach ($results as $entity) {
            $lastId = $entity->id;

            yield $entity;
        }
    }
}
```

Reply:
I used your code and changed just a bit, to have type iteration (reindexing by type order, even when no arguments are passed to the command).

Funny note: your `objectIterator` realizes an infinite loop, because results is fetched with the wrong variable (`$query` instead of `$q`); it took me some time to "see" it :-)